### DISCUSSION TITLE
hello,How to set transaction isolation level

### DISCUSSION THREAD

**Discussion by bitszhang** [duckdb/duckdb#12992] [Q&A]
When I was updating the data in the table, I used a transaction. Before my transaction was committed, other processes could not read the table. So, how can I change the transaction isolation level to REPEATABLE READ

**Comment by bitszhang** [comment]
Can anyone help me? Thank you

**Reply by Tishj** [reply]
I don't think we have isolation levels like that.
Any in-flight transaction is invisible to other in-flight transactions

**Reply by bitszhang** [reply]
I tested again and found that without using transactions, such as executing DELETE FROM TABLE, other requests would be blocked until the delete operation was completed. How can this phenomenon be handled so that it does not affect each other? Looking forward to your answer. Thank you

**Reply by Tishj** [reply]
I'm not sure I follow, can you provide some code to clarify this?

I am going to assume you're talking about executing statements concurrently.
This is possible, but you'll need to make a separate connection to the db for each statement.
Only one execution can be active per connection at a time

**Reply by bitszhang** [reply]
After your reminder, I checked the Python duck db package, cursor (self: duck db. duck db.) DuckDBPyConnection) ->duck db. duck db. DuckDBPyConnection can create connection replicas, but there is no such method in the go duck db package, which results in my query not being a new connection every time. Do you have any suggestions

**Reply by Tishj** [reply]
The go package wraps the C API, so this is definitely supported.

`cursor` is python specific, just to conform to DBAPI.

You want to create a new connection from the same database to get the same behavior

**Reply by bitszhang** [reply]
This is my duckdb struct. I call GetInstance once in the main function. All goroutines use the same db, which will cause the service to be blocked. Now I don't know how to change it. Could you give me some guidance? Thank you very much

```
package main
import (
	"database/sql"
	"fmt"
	"log"
	"net/http"
	"sync"

	"github.com/gin-gonic/gin"
	_ "github.com/marcboeker/duckdb"
)

type DuckDB struct {
	dataSourceName string
	db             *sql.DB
}

var (
	instance *DuckDB
	once     sync.Once
)

// NewDuckDB 创建一个新的 DuckDB 实例，并配置连接池
func NewDuckDB(dataSourceName string) (*DuckDB, error) {
	db, err := sql.Open("duckdb", dataSourceName)
	if err != nil {
		return nil, err
	}

	// 配置连接池参数
	db.SetMaxOpenConns(10)  // 最大打开连接数
	db.SetMaxIdleConns(5)   // 最大空闲连接数
	db.SetConnMaxLifetime(0) // 连接可复用的最大时间，0表示无限制

	return &DuckDB{
		dataSourceName: dataSourceName,
		db:             db,
	}, nil
}

// GetInstance 返回 DuckDB 的全局实例
func GetInstance(dataSourceName string) (*DuckDB, error) {
	var err error
	once.Do(func() {
		instance, err = NewDuckDB(dataSourceName)
	})
	return instance, err
}

// CreateTable 创建表
func (d *DuckDB) CreateTable(query string) error {
	_, err := d.db.Exec(query)
	return err
}

// ExecuteQuery 执行查询并返回结果
func (d *DuckDB) ExecuteQuery(query string) (*sql.Rows, error) {
	return d.db.Query(query)
}
//执行删除操作
func (d *DuckDB) Delete(query string) error {
	_, err := d.Db.Exec(query)
	return err
}
```

**Reply by bitszhang** [reply]
Just to add, my data volume is about 50000 entries, and the data table contains several indexes. When executing delete from table, it takes about 15 seconds. During these 15 seconds, all other read and write operations are waiting, but I removed the indexes. It seems that the problem has been solved and the performance seems to be faster
